test/suite/test_rollback_to_stable16.py

```diff
@@ -0,0 +1,130 @@
+class test_rollback_to_stable16(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=200MB,statistics=(all)'
+    session_config = 'isolation=snapshot'
+
+    def simulate_crash_restart(self, olddir, newdir):
+        ''' Simulate a crash from olddir and restart in newdir. '''
+        # with the connection still open, copy files to new directory
+        shutil.rmtree(newdir, ignore_errors=True)
+        os.mkdir(newdir)
+        for fname in os.listdir(olddir):
+            fullname = os.path.join(olddir, fname)
+            # Skip lock file on Windows since it is locked
+            if os.path.isfile(fullname) and \
+                "WiredTiger.lock" not in fullname and \
+                "Tmplog" not in fullname and \
+                "Preplog" not in fullname:
+                shutil.copy(fullname, newdir)
+        #
+        # close the original connection and open to new directory
+        # NOTE:  This really cannot test the difference between the
+        # write-no-sync (off) version of log_flush and the sync
+        # version since we're not crashing the system itself.
+        #
+        self.close_conn()
+        self.conn = self.setUpConnectionOpen(newdir)
+        self.session = self.setUpSessionOpen(self.conn)
+
+    def insert_update_data_at_given_timestamp(self, uri, value, start_row, end_row, timestamp):
+        cursor =  self.session.open_cursor(uri)
+        for i in range(start_row, end_row):
+            self.session.begin_transaction()
+            cursor[i] = value + str(i)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(timestamp))
+        cursor.close()
+
+    def check(self, check_value, uri, nrows, start_row, end_row, read_ts):
+        session = self.session
+        session.begin_transaction('read_timestamp=' + timestamp_str(read_ts))
+        cursor = session.open_cursor(uri)
+
+        count = 0
+        for i in range(start_row, end_row):
+            cursor.set_key(i)
+            ret = cursor.search()
+            # self.tty(f'value = {cursor.get_value()}')
+            if check_value is None:
+                if ret != wiredtiger.WT_NOTFOUND:
+                    self.tty(f'value = {cursor.get_value()}')
+                    self.assertTrue(ret == wiredtiger.WT_NOTFOUND)
+            else:
+                self.assertEqual(cursor.get_value(), check_value + str(count + start_row))
+                count += 1
+        session.commit_transaction()
+        # self.assertEqual(count, nrows)
+        cursor.close()
+
+    def test_rollback_to_stable16(self):
+        # Create a table.
+        uri = "table:rollback_to_stable16"
+        nrows = 200
+        create_params = 'key_format=r,value_format=S'
+
+        self.session.create(uri, create_params)
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        value20 = "aaaa"
```

Comment:
The variable names for the values are a bit odd. Maybe consider writing this in a loop? Something like this (please forgive python syntax errors):

```
values = ['aaaa', 'bbbb', 'cccc', 'dddd']
ts = [2*k+1 for k in range(1, len(values)+1)]
start_row = 1
for i in range (0, len(values)+1):
    self.insert_update_data_at_given_timestamp(uri, values[i],  start_row, nrows, ts[i])
    start_row = start_row + nrows
```